server/aws-lsp-codewhisperer/src/language-server/chat/telemetry/chatTelemetryController.ts

```diff
@@ -54,13 +55,15 @@ export class ChatTelemetryController {
     #activeTabId?: string
     #tabTelemetryInfoByTabId: { [tabId: string]: ConversationTriggerInfo }
     #currentTriggerByTabId: { [tabId: string]: TriggerType } = {}
+    #customizationInfoByTabAndMessageId: { [tabId: string]: { [messageId: string]: string } }
```

Comment:
Would be using only `messageId` to store this value be enough? If `messageId` is unique uuid across all tabs, then grouping per tab is redundant

Reply:
Very good question, I also considered that but storing by message id means we have no way of knowing when a message is no longer active/relevant. 

When we store messages inside their respective tabs, then we can safely clean up all the messages that are no longer active by deleting the tab entry. So this is to ensure the `customizationInfoByTabAndMessageId` doesn't keep growing infinitely